src/SFML/Window/Unix/WindowImplX11.cpp

```diff
@@ -1061,54 +1063,128 @@ void WindowImplX11::setVideoMode(const VideoMode& mode)
+    if (!res)
     {
-        // Failed to get the screen configuration
-        err() << "Failed to get the current screen configuration for fullscreen mode, switching to window mode" << std::endl;
+        // Failed to get screen resources
+        err() << "Failed to get the current screen resources for fullscreen mode, switching to window mode" << std::endl;
         return;
     }
 
-    // Save the current video mode before we switch to fullscreen
-    Rotation currentRotation;
-    m_oldVideoMode = XRRConfigCurrentConfiguration(config, &currentRotation);
+    // If version > 1.2 get the primary screen else take the first screen
+    RROutput output;
+    if ((xRandRMajor == 1 && xRandRMinor >= 3) || xRandRMajor > 1)
+    {
+        output = XRRGetOutputPrimary(m_display, rootWindow);
+
+        // Check if output return is valid, otherwise use the first screen
+        if(output == None)
+            output = res->outputs[0];
+    }
+    else
+    {
+        output = res->outputs[0];
+    }
 
-    // Get the available screen sizes
-    int nbSizes;
-    XRRScreenSize* sizes = XRRConfigSizes(config, &nbSizes);
+    // Get output info from output
+    XRROutputInfo* outputInfo = XRRGetOutputInfo(m_display, res, output);
+    if (!outputInfo || outputInfo->connection == RR_Disconnected)
+    {	
+        // Free screen resources
+        XRRFreeScreenResources(res);
+        
+        // If outputInfo->connection == RR_Disconnected, free output info
+        if(!outputInfo)
+        {
+            XRRFreeOutputInfo(outputInfo);
+        }
 
-    // Search for a matching size
-    for (int i = 0; (sizes && i < nbSizes); ++i)
+        // Failed to get output info
+        err() << "Failed to get output info for fullscreen mode, switching to window mode" << std::endl;
+        return;
+    }
+
+    // Get crtc info from crtc
+    XRRCrtcInfo* crtcInfo = XRRGetCrtcInfo(m_display, res, outputInfo->crtc);
+    if (!crtcInfo)
     {
-        XRRConfigRotations(config, &currentRotation);
+        // Free screen resources
+        XRRFreeScreenResources(res);
 
-        if (currentRotation == RR_Rotate_90 || currentRotation == RR_Rotate_270)
-            std::swap(sizes[i].height, sizes[i].width);
+        // Free  output info
+        XRRFreeOutputInfo(outputInfo);
 
-        if ((sizes[i].width  == static_cast<int>(mode.width)) && (sizes[i].height == static_cast<int>(mode.height)))
-        {
-            // Switch to fullscreen mode
-            XRRSetScreenConfig(m_display, config, RootWindow(m_display, m_screen), i, currentRotation, CurrentTime);
+        // Failed to get crtc info
+        err() << "Failed to get crtc info for fullscreen mode, switching to window mode" << std::endl;
+        return;
+    }
+
+    // Find RRMode to set 
+    bool modeFound = false;
+    RRMode xRandMode;
 
-            // Set "this" as the current fullscreen window
-            fullscreenWindow = this;
+    for (int i = 0; i < res->nmode; i++)
```

Comment:
I know it was like that before, but it would be much more elegant IMO to have `i < res->nmode && !modeFound` as a loop condition and remove the `break statement`, wouldn't it?